cli/tests/unit_node/_fs/_fs_writeFile_test.ts

```diff
@@ -0,0 +1,346 @@
+    },
+    Error,
+    `The value "made-up-encoding" is invalid for option "encoding"`,
+  );
+
+  assertThrows(
+    () => {
+      writeFile(
+        "some/path",
+        "some data",
+        {
+          // @ts-expect-error Type '"made-up-encoding"' is not assignable to type
+          encoding: "made-up-encoding",
+        },
+        () => {},
+      );
+    },
+    Error,
+    `The value "made-up-encoding" is invalid for option "encoding"`,
+  );
+
+  assertThrows(
+    () => {
+      writeFileSync("some/path", "some data", {
+        // @ts-expect-error Type '"made-up-encoding"' is not assignable to type
+        encoding: "made-up-encoding",
+      });
+    },
+    Error,
+    `The value "made-up-encoding" is invalid for option "encoding"`,
+  );
+});
+
+Deno.test(
+  "Unsupported encoding results in error()",
+  function testUnsupportedEncoding() {
+    assertThrows(
+      () => {
+        writeFile("some/path", "some data", "utf16le", () => {});
+      },
+      Error,
+      `Not implemented: "utf16le" encoding`,
+    );
+
+    assertThrows(
+      () => {
+        writeFileSync("some/path", "some data", "utf16le");
+      },
+      Error,
+      `Not implemented: "utf16le" encoding`,
+    );
+  },
+);
+
+Deno.test(
+  "Data is written to correct rid",
+  async function testCorrectWriteUsingRid() {
+    const tempFile: string = await Deno.makeTempFile();
+    const file: Deno.FsFile = await Deno.open(tempFile, {
+      create: true,
+      write: true,
+      read: true,
+    });
+
+    await new Promise<void>((resolve, reject) => {
+      writeFile(file.rid, "hello world", (err) => {
+        if (err) return reject(err);
+        resolve();
+      });
+    });
+    Deno.close(file.rid);
+
+    const data = await Deno.readFile(tempFile);
+    await Deno.remove(tempFile);
+    assertEquals(decoder.decode(data), "hello world");
+  },
+);
+
+Deno.test(
+  "Data is written to correct file",
+  async function testCorrectWriteUsingPath() {
+    const res = await new Promise((resolve) => {
+      writeFile("_fs_writeFile_test_file.txt", "hello world", resolve);
+    });
+
+    const data = await Deno.readFile("_fs_writeFile_test_file.txt");
+    await Deno.remove("_fs_writeFile_test_file.txt");
+    assertEquals(res, null);
+    assertEquals(decoder.decode(data), "hello world");
+  },
+);
+
+Deno.test(
+  "Data is written to correct file encodings",
+  async function testCorrectWriteUsingDifferentEncodings() {
+    const encodings = [
+      ["hex", "68656c6c6f20776f726c64"],
+      ["HEX", "68656c6c6f20776f726c64"],
+      ["base64", "aGVsbG8gd29ybGQ="],
+      ["BASE64", "aGVsbG8gd29ybGQ="],
+      ["utf8", "hello world"],
+      ["utf-8", "hello world"],
+    ];
+
+    for (const [encoding, value] of encodings) {
+      const res = await new Promise((resolve) => {
+        writeFile(
+          "_fs_writeFile_test_file.txt",
+          value,
+          encoding as TextEncodings,
```

Comment:
What happens if you just skip `as TextEncodings` bit?

Reply:
It complais: `Argument of type 'string' is not assignable to parameter of type 'WriteFileOptions'`. This parameter should be `Encodings | CallbackWithError | WriteFileOptions | undefined`, with type `Encodings = BinaryEncodings | TextEncodings`.